NorthstarDLL/logging/logging.cpp

```diff
@@ -220,3 +223,70 @@ void NS::log::FlushLoggers()
 
 	spdlog::default_logger()->flush();
 }
+
+// Wine specific functions
+typedef const char*(CDECL* wine_get_host_version_type)(const char**, const char**);
+wine_get_host_version_type wine_get_host_version;
+
+typedef const char*(CDECL* wine_get_build_id_type)(void);
+wine_get_build_id_type wine_get_build_id;
+
+// Not exported Winapi methods
+typedef NTSTATUS (WINAPI* RtlGetVersion_type)(PRTL_OSVERSIONINFOW);
+RtlGetVersion_type RtlGetVersion;
+
+void StartupLog()
+{
+	spdlog::info("NorthstarLauncher version: {}", version);
+	spdlog::info("Command line: {}", GetCommandLineA());
+	spdlog::info("Using profile: {}", GetNorthstarPrefix());
+
+	HMODULE ntdll = GetModuleHandleA("ntdll.dll");
+	if (!ntdll)
+	{
+		// How did we get here
+		spdlog::info("Operating System: Unknown");
+		return;
+	}
+
+	wine_get_host_version = (wine_get_host_version_type)GetProcAddress(ntdll, "wine_get_host_version");
+	if (wine_get_host_version)
+	{
+		// Load the rest of the functions we need
+		wine_get_build_id = (wine_get_build_id_type)GetProcAddress(ntdll, "wine_get_build_id");
+
+		const char* sysname;
+		wine_get_host_version(&sysname, NULL);
+
+		spdlog::info("Operating System: {} (Wine)", sysname);
+		spdlog::info("Wine build: {}", wine_get_build_id());
+
+		char* compatToolPtr = std::getenv("STEAM_COMPAT_TOOL_PATHS");
+		if (compatToolPtr)
+		{
+			std::string compatToolPath(compatToolPtr);
+
+			spdlog::info("Proton build: {}", compatToolPath.substr(compatToolPath.rfind("/") + 1));
+		}
+	}
+	else
+	{
+		// We are real Windows (hopefully)
+		const char* win_ver = "Unknown";
+
+		RTL_OSVERSIONINFOW osvi;
+		osvi.dwOSVersionInfoSize = sizeof(osvi);
+
+		RtlGetVersion = (RtlGetVersion_type)GetProcAddress(ntdll, "RtlGetVersion");
+		if (RtlGetVersion && RtlGetVersion(&osvi))
```

Comment:
`STATUS_SUCCESS` ( What `RtlGetVersion` returns ) is 0x0 ( [docs](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55) ). This means this check will fail and fall back to the else block.

Modifying the check for the return value yielded this:
![image](https://github.com/R2Northstar/NorthstarLauncher/assets/64418963/0133dee9-fd1f-4f3e-80c2-da8386023513)

Reply:
Yep, I knew this but ended up fumbling it up along the way.
It appears to work as expected, so I guess thats solved.